lib/hsa/mcwamp_hsa.cpp

```diff
@@ -4896,11 +4896,27 @@ HSABarrier::waitComplete() {
 // TODO - remove hsaQueue parm.
 inline hsa_status_t
 HSABarrier::enqueueAsync(hc::memory_scope fenceScope) {
+    hc::memory_scope toDoReleaseFenceScope = fenceScope;
+    hc::memory_scope toDoAcquireFenceScope = fenceScope;
 
-    if (fenceScope == hc::system_scope) {
-        hsaQueue()->setNextSyncNeedsSysRelease(false);
+    hsa_queue_t* rocrQueue = hsaQueue()->acquireLockedRocrQueue();
+
+    if ( hsaQueue()->nextSyncNeedsSysRelease() ) {
+         toDoReleaseFenceScope = hc::system_scope;
+         DBOUTL( DB_CMD2, "absorb System Scope Release Fencing request from the HSAQueue") ;
+    };
+
+    if ( toDoReleaseFenceScope == hc::system_scope )
+         hsaQueue()->setNextSyncNeedsSysRelease(false);
```

Comment:
Should be moved into the `if` branch above since it only needs to be set to `false` when the original value is `true`